dongle/mt76.cpp

```diff
@@ -1408,6 +1408,31 @@ bool Mt76::initChannels()
     return true;
 }
 
+bool Mt76::openFirmware()
+{
+    const std::string filename(DATADIR "/FW_ACC_00U.bin");
+
+    if (firmware.size() != 0)
```

Comment:
firmware.size() calls std::vector<uint8_t>.size, and the vector shouldn't have any data in it so this checks seems pointless to me.

But if there is reason maybe adopting std::vector.empty() is worthwhile

Reply:
`firmware::data` aka `Bytes::data` is private. Adding `::empty()` accessor for this is an overkill IMHO.

Reply:
possibly, but is this check even needed? shouldn't firmware be empty when we reach this?

Reply:
IIRC the `loadFirmware()` code suggest that it could be called more than once, which ultimately call this function. Or that's not the case? Happy to drop it if so.

Reply:
`loadFirmware` is only called in the constructor, so you can drop that check. I also see no point in having a `Mt76::firmware` attribute since the firmware is only needed by the `loadFirmware` function.

Reply:
Even better thanks.